package.py

```diff
@@ -914,6 +955,121 @@ def install_pip_requirements(query, requirements_file):
+    # TODO:
+    #  1. Emit files instead of temp_dir
+    poetry_lock_file = os.path.join(path, "poetry.lock")
+    pyproject_file = os.path.join(path, "pyproject.toml")
+    # pyproject.toml is always required by poetry, poetry.lock is optional and can be created from pyproject.toml
+    if not os.path.exists(pyproject_file):
+        yield
+        return
+
+    runtime = query.runtime
+    artifacts_dir = query.artifacts_dir
+    docker = query.docker
+    docker_image_tag_id = None
+
+    if docker:
+        docker_file = docker.docker_file
+        docker_image = docker.docker_image
+        docker_build_root = docker.docker_build_root
+
+        if docker_image:
+            ok = False
+            while True:
+                output = check_output(docker_image_id_command(docker_image))
+                if output:
+                    docker_image_tag_id = output.decode().strip()
+                    log.debug(
+                        "DOCKER TAG ID: %s -> %s", docker_image, docker_image_tag_id
+                    )
+                    ok = True
+                if ok:
+                    break
+                docker_cmd = docker_build_command(
+                    build_root=docker_build_root,
+                    docker_file=docker_file,
+                    tag=docker_image,
+                )
+                check_call(docker_cmd)
+                ok = True
+        elif docker_file or docker_build_root:
+            raise ValueError(
+                "docker_image must be specified " "for a custom image future references"
+            )
+
+    working_dir = os.getcwd()
+
+    log.info("Installing python dependencies with poetry: %s", poetry_lock_file)
+    with tempdir() as temp_dir:
+        def copy_file_to_target(file, temp_dir):
+            filename = os.path.basename(file)
+            target_file = os.path.join(temp_dir, filename)
+            shutil.copyfile(file, target_file)
+            return target_file
+
+        poetry_lock_target_file = copy_file_to_target(poetry_lock_file, temp_dir)
+        pyproject_target_file = copy_file_to_target(pyproject_file, temp_dir)
+
+        poetry_exec = "poetry"
+        subproc_env = None
+
+        if not docker:
+            if WINDOWS:
+                poetry_exec = "poetry.bat"
+
+        # Install dependencies into the temporary directory.
+        with cd(temp_dir):
+            # NOTE: poetry must be available, which is the case with lambci/lambda:build-python* docker images
+            # FIXME: poetry install does not currently allow to specify the target directory so we extract
+            # installed packages from .venv/lib/python*/site-packages
+            poetry_commands = [
+                shlex_join([ poetry_exec, "config", "--no-interaction", "virtualenvs.create", "true" ]),
+                shlex_join([ poetry_exec, "config", "--no-interaction", "virtualenvs.in-project", "true" ]),
+                shlex_join([ poetry_exec, "install", "--no-interaction" ]),
```

Comment:
Until `poetry install` supports a `--target` parameter, an alternative may be to let [poetry export](https://python-poetry.org/docs/cli/#export) the dependencies in `requirements.txt` format then install dependencies with `pip`:

```suggestion
                shlex_join([ poetry_exec, "-f", "requirements.txt", "--output", "requirements.txt" ]),
                shlex_join([ python_exec, '-m', 'pip', 'install', '--no-compile', '--prefix=', '--target=.','--requirement=requirements.txt'),

```

Reply:
@pdecat please add `--no-dev` to the install command. I believe it's more correct behaviour

Reply:
@Tonkonozhenko good point, done!